pyhf/paramsets.py

```diff
@@ -37,41 +37,55 @@ def expected_data(self, pars):
         )
 
 
-def reduce_paramset_requirements(paramset_requirements):
-    reduced_paramset_requirements = {}
+def reduce_paramsets_requirements(paramsets_requirements, paramsets_user_configs):
+    reduced_paramsets_requirements = {}
 
-    # nb: normsys and histosys have different op_codes so can't currently be shared
-    param_keys = [
+    paramset_keys = [
         'paramset_type',
         'n_parameters',
-        'op_code',
         'inits',
         'bounds',
         'auxdata',
         'factors',
     ]
 
-    for param_name in list(paramset_requirements.keys()):
-        params = paramset_requirements[param_name]
+    for paramset_name in list(paramsets_requirements.keys()):
```

Comment:
why are we looping over `paramsets_requirements.keys()` and not `paramset_keys` ?